New Public Team Server API update

platrofa opened the pull request.

Commit:
New Public Team Server API update

ahmed-negm commented on content/apidocs-mxsdk/apidocs/old-version/team-server-api.md:
```diff
@@ -0,0 +1,206 @@
+---
```
Let's check if we need to keep the old version.

platrofa replied:
I added a warning on top that informs the reader about the plan to deprecate the current API and links to the new API

platrofa replied:
I created a new link in the index page about the APIs. Now there is a link to the `normal` API and a second one that points to the `previous` implementation. The `previous` implementations now shows a warning about the incoming deprecation and links the new APIs.

ahmed-negm replied:
I would add a notice on top of the new documentation that links to the old version. People having bookmarked the current documentation may otherwise wonder where it went.

platrofa replied:
Added the new notice in the new documentation that links to the old version.

ahmed-negm commented on content/howto/integration/implement-cicd-pipeline.md:
```diff
@@ -53,7 +53,7 @@ This is a POST call, so you will need to pass the relevant fields in the body (f
 ![](attachments/implement-cicd-pipeline/03StartBuildingPackage.png)
 
-Before proceeding to the next step, you need to wait for the build of the deployment package to be successful. To do this, use the [Retrieve Package API call](/apidocs-mxsdk/apidocs/build-api#retrieve-package). This scenario needs to use the `PackageId` (output from the previous call) and check if the status is `Succeeded` (the other possible statuses are Queued, Building, Uploading, and Failed).
+Before proceeding to the next step, you need to wait for the build of the deployment package to be successful. To do this, use the [Retrieve Package API call](/apidocs-mxsdk/apidocs/old-version/build-api#retrieve-package). This scenario needs to use the `PackageId` (output from the previous call) and check if the status is `Succeeded` (the other possible statuses are Queued, Building, Uploading,  ...
```
We didn't change `build-api` page

platrofa replied:
Removed the `old-version` folder as we want to have both versions side by side during the transition period

platrofa replied:
Oops. Reverted.

ahmed-negm commented on content/howto7/integration/implement-cicd-pipeline.md:
```diff
@@ -39,7 +39,7 @@ To build the deployment package, use the [Team Server API](/apidocs-mxsdk/apidoc
 #### 3.2.1 Geting the Latest Revision
 
-To get the latest revision, use the [Retrieve Branch API call](/apidocs-mxsdk/apidocs/team-server-api#retrieve-branch) to get the `LatestRevisionNumber` from the output.
+To get the latest revision, use the [Retrieve Branch API call](/apidocs-mxsdk/apidocs/old-version/team-server-api#retrieve-branch) to get the `LatestRevisionNumber` from the output.
```
This should use the new documentation. AFAIK, the old APIs will be deprecated once we make ours public.

platrofa replied:
The link now points to the new APIs

ahmed-negm commented on content/howto8/integration/implement-cicd-pipeline.md:
```diff
@@ -39,7 +39,7 @@ To build the deployment package, use the [Team Server API](/apidocs-mxsdk/apidoc
 #### 3.2.1 Getting the Latest Revision
 
-To get the latest revision, use the [Retrieve Branch API call](/apidocs-mxsdk/apidocs/team-server-api#retrieve-branch) to get the `LatestRevisionNumber` from the output.
+To get the latest revision, use the [Retrieve Branch API call](/apidocs-mxsdk/apidocs/old-version/team-server-api#retrieve-branch) to get the `LatestRevisionNumber` from the output.
```
Same comment, let's use the new documentation.

platrofa replied:
The link now points to the new APIs

platrofa replied:
Removed

ahmed-negm commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -6,201 +6,444 @@ menu_order: 65
+Error Response format and examples are given in section: [Error response format](#3-error-response-format)
 
-##### 3.3.2.2 Example
+## 5 Webhook for new commits
```
Webhooks is not part of TS:Git GA, we shouldn't publish this section.

platrofa replied:
Removed

Commit:
Review: Apply review comments

bvanreeven commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -6,201 +6,409 @@ menu_order: 65
+The team server API allows you to retrieve the information (branches, commits) of application models stored in our team server. You always access an application model via the context of an application (see the Deploy API for more information about retrieving applications and application identifiers).
 
 {{% alert type="warning" %}}
 The team server API is only available to *licensed* apps which are running in a Mendix Cloud.
```
This warning does not apply to the new API.

platrofa replied:
Removed

bvanreeven commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
 ---
 
+{{% alert type="warning" %}}
+The API description on this page refers to the new team server API implementation. You can access the previous team server API implementation [here](old-version/team-server-api.md).
```
Let's clarify it a bit as follows (new stuff in _italic_):

You can access _the documentation for_ the previous team server API implementation ...

platrofa replied:
I agree. It was not so clear before.

bvanreeven commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
-The team server API allows you to retrieve the information (branches, revisions) of application models stored in our team server. You always access an application model via the context of an application (see the Deploy API for more information about retrieving applications and application identifiers).
-The image below provides a domain model representation of the concepts discussed below and how these are related:
-![](attachments/131076/425989.png)
+The team server API allows you to retrieve the information (branches, commits) of application models stored in our team server. You always access an application model via the context of an application (see the Deploy API for more information about retrieving applications and application identifiers).
```
> You always access an application model via the context of an application (see the Deploy API for more information about retrieving applications and application identifiers).

This statement does not make sense for the new API. Let's remove it.

platrofa replied:
Removed

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/old-version/team-server-api.md:
```diff
@@ -0,0 +1,210 @@
+
+The team server API allows you to retrieve the information (branches, revisions) of application models stored in our team server. You always access an application model via the context of an application (see the Deploy API for more information about retrieving applications and application identifiers).
+The image below provides a domain model representation of the concepts discussed below and how these are related:
+![](attachments/131076/425989.png)
```
We use an absolute path for cross-reference, for example, here it's better to change it to:`(/apidocs-mxsdk/apidocs/team-server-api)`.

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/old-version/team-server-api.md:
```diff
@@ -0,0 +1,210 @@
+
+The team server API allows you to retrieve the information (branches, revisions) of application models stored in our team server. You always access an application model via the context of an application (see the Deploy API for more information about retrieving applications and application identifiers).
+The image below provides a domain model representation of the concepts discussed below and how these are related:
+![](attachments/131076/425989.png)
```
The attachment file path needs to be changed, too, because the markdown file is now in a different folder. Is it possible to change the name of this old API doc to `team-server-api-version-xxx` and leave it in the same folder, instead of putting this old API doc in a new folder?

Luyao-Zhang-1 replied:
Is it possible to mention a version number instead of just calling it "old version"? In the future, if we have newer versions, then the current "new version" would become an "old version", too, and this "old version" would become an "even older version". It would save us troubles if we could just give it a version number now.

platrofa replied:
We reverted the changes on the `Team Server API`. The new API will be called `App Repository API`. A versioning is no longer needed as now they are two different APIs.

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
 ---
 
+{{% alert type="warning" %}}
+The API description on this page refers to the new team server API implementation. You can access the documentation for the previous team server API implementation [here](old-version/team-server-api.md).
```
Same as above – if possible, use version numbers.

Luyao-Zhang-1 replied:
`here` should be changed to the name of the doc. (`For information about the implementation of team server API version xxx , see [Team Server API Version XXX](team-server-api-version-xxx).`)

platrofa replied:
Removed warning as we will call this new API `App Repository API`

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
-{{% alert type="warning" %}}
-The team server API is only available to *licensed* apps which are running in a Mendix Cloud.
-{{% /alert %}}
+The base URL for all Team Server API endpoints is:
```
Can we make the base URL a separate section to make it stand out?

platrofa replied:
`Introduction`, `Base URL` and `Endpoints` are now separate sections

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
-The Developer Portal Management API requires its users to authenticate themselves. This can be done by using API keys; for more information about this please refer to [this article](authentication "authentication").
+The Team Server API requires a Personal Access Token (PAT) as authentication method.
+
+You can manage your Mendix personal access tokens via [warden](https://warden.mendix.com/). Press "add", then enter the token name and choose at least the following scope: `mx:modelrepository:repo:read`
```
Press **Add**, then enter the token **Name**..., and then click **Create**.

platrofa replied:
Done

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
+
+| Name | Type | Description |
+| --- | --- | --- |
+| `errorCode` | String | A code that can be used to lookup the error. |
```
lookup -> look up

platrofa replied:
Thank you for noticing. Fixed.

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
+All available endpoints are:
+| Method | Endpoint | Description |
+| --- | --- | --- |
+| GET | [`/repositories/<projectId>`](#41-retrieve-repository-info) | Get repository information for a project |
```
Need to add anchors {#XXX} manually to the linked sections, and use the added anchors here for cross-references.

platrofa replied:
Added the missing manual anchors

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
+| Method | Endpoint | Description |
+| --- | --- | --- |
+| GET | [`/repositories/<projectId>`](#41-retrieve-repository-info) | Get repository information for a project |
+| GET | [`/repositories/<projectId>/branches`](#42-retrieve-branches) | Get a list of branches of a repository |
```
Here is `projectId`, but in the sections below it says `AppId`. It is better to keep it consistent at all places.

bvanreeven replied:
Indeed, let's use `appId` everywhere.

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
+|`projectId`|String|The project ID (also known as app ID) of the Mendix project.|
+|`type`|String|The type of repository. At the moment this will be either `"svn"` or `"git"`, but later on other repository types may be introduced.|
+|`url`|String|The URL of the repository.|
+|`isMendixHosted`|Boolean|Whether the repository is Mendix-hosted (i.e. a Team Server repository) or is using a custom repository URL. Note: If a repository is not Mendix-hosted (i.e. `isMendixHosted` is `false`), the other operations of the Team Server API will not work for this repository.|
```
Apply the note style:
{{% alert type="note" %}}
If a repository is not Mendix-hosted (i.e. `isMendixHosted` is `false`), the other operations of the Team Server API will not work for this repository.
{{% /alert %}}

platrofa replied:
We removed `isMendixHosted` field from the API, this warning is now removed from the docs

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
+| 404 | Not Found | Repository not found |
+| 500 | Internal Server Error | Something went wrong |
+
+Error Response format and examples are given in section: [Error response format](#3-error-response-format)
```
Need to add anchors {#XXX} manually to the linked sections, and use the added anchors here for cross-references.

platrofa replied:
Done

Luyao-Zhang-1 commented on content/apidocs-mxsdk/apidocs/team-server-api.md:
```diff
@@ -4,203 +4,411 @@ category: "API Documentation"
+|Name|Type|Required|Description
+|---|---|---|---|
+|`limit`|Integer|No|The number of items to return per result page. Defaults to 20 items. Maximum is 100 items.|
+|`cursor`|String|No|A cursor specifying which page to retrieve. To obtain a cursor value, see the `cursors` property of the response payload of this operation. If no cursor is specified, the first page is returned. Note: To the user of this API, a cursor is an opaque value that can only be obtained from a previous API response.|
```
Use the note style:
{{% alert type="note" %}}
To the user of this API, a cursor is an opaque value that can only be obtained from a previous API response.
{{% /alert %}}

platrofa replied:
Style applied.

bvanreeven committed:
Update Team Server API docs for new endpoint

- Replace endpoint for retrieving repository info with new /info endpoint
- Update terminology: project -> app, projectId -> appId
- Fix broken links to/from old documentation
- Fix table of endpoints and broken links to endpoint sections

Commit:
Rename `New Public Team Server API` to `App Repository API`

Commit:
Restore `Team Server API`

Commit:
Deprecate Team Server API

Commit:
Add `App Repository API` to apidocs index

Commit:
Review: Separate `Base URL` and `Endpoints` sections

Commit:
Review: Highlight PAT creation steps

Commit:
Review: Typo: `lookup` -> `look up`

Commit:
Review: Add manual ancors and use them for cross-references

Commit:
Review: Apply note style to opaque cursor

bvanreeven commented on content/apidocs-mxsdk/apidocs/app-repository-api.md:
```diff
@@ -0,0 +1,410 @@
+
+## 1 Introduction
+
+The App Repository API enables retrieving information (branches, commits) of application models stored in our team server.
```
'team server' --> 'Team Server'

Also, let's make it a link to the Team Server documentation page (in `developerportal/collaborate/team-server`)

platrofa replied:
Done.

Commit:
Review: Add Team Server link

bvanreeven committed:
Improve table format for API response documentation

Commit:
Order `API Documentation` alphabetically

Commit:
Fix menu order
